pallets/ema-oracle/src/lib.rs

```diff
@@ -299,11 +316,42 @@ pub mod pallet {
 		pub fn update_bifrost_oracle(
 			origin: OriginFor<T>,
 			//NOTE: these must be boxed becasue of https://github.com/paritytech/polkadot-sdk/blob/6875d36b2dba537f3254aad3db76ac7aa656b7ab/substrate/frame/utility/src/lib.rs#L150
-			_asset_a: Box<polkadot_xcm::VersionedLocation>,
-			_asset_b: Box<polkadot_xcm::VersionedLocation>,
-			_price: (Balance, Balance),
+			asset_a: Box<polkadot_xcm::VersionedLocation>,
+			asset_b: Box<polkadot_xcm::VersionedLocation>,
+			price: (Balance, Balance),
 		) -> DispatchResult {
 			T::BifrostOrigin::ensure_origin(origin)?;
+
+			let asset_a_loc = Location::try_from(*asset_a).map_err(|()| Error::<T>::BadVersion)?;
+			let asset_b_loc = Location::try_from(*asset_b).map_err(|()| Error::<T>::BadVersion)?;
+
+			let asset_a = T::CurrencyIdConvert::convert(asset_a_loc).ok_or(Error::<T>::AssetNotFound)?;
+			let asset_b = T::CurrencyIdConvert::convert(asset_b_loc).ok_or(Error::<T>::AssetNotFound)?;
```

Comment:
hm.. why the need of two conversion ? 

i would epxect to have just passed the given param into the conversion function, and it should handle it

Reply:
not sure what yuu mean, the trait is Convert where the implementation can only convert one location at a time.